### ISSUE TITLE
java.lang.IllegalStateException: Not allowed to start service Intent

### ISSUE THREAD

**Issue by bshka** [OneSignal/OneSignal-Android-SDK#624]
**Description:**
App craches on Android 9

**Environment**
Device - Pixel 2 XL Android 9

**Steps to Reproduce Issue:**

1. Start app
2. Just hide and reopen it

```sh
classpath 'gradle.plugin.com.onesignal:onesignal-gradle-plugin:[0.11.2, 0.99.99]'
implementation 'com.onesignal:OneSignal:[3.7.1, 3.99.99]'
```

**Anything else:**

```sh
Caused by java.lang.IllegalStateException: Not allowed to start service Intent { cmp=social.heyya/com.onesignal.SyncService (has extras) }: app is in background uid UidRecord{b15da14 u0a248 TPSL bg:+1m57s616ms idle change:cached procs:1 seq(0,0,0)}
       at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1577)
       at android.app.ContextImpl.startService(ContextImpl.java:1532)
       at android.content.ContextWrapper.startService(ContextWrapper.java:664)
       at com.onesignal.OneSignal.startSyncService(OneSignal.java:2258)
       at com.onesignal.OneSignal.onAppFocus(OneSignal.java:947)
       at com.onesignal.ActivityLifecycleHandler.handleFocus(ActivityLifecycleHandler.java:120)
       at com.onesignal.ActivityLifecycleHandler.onActivityResumed(ActivityLifecycleHandler.java:74)
       at com.onesignal.ActivityLifecycleListener.onActivityResumed(ActivityLifecycleListener.java:48)
       at android.app.Application.dispatchActivityResumed(Application.java:239)
       at android.app.Activity.onResume(Activity.java:1343)
       at android.support.v4.app.FragmentActivity.onResume(FragmentActivity.java:485)
       at com.heyya.installed.activity.abstracts.GoogleServicesActivity.onResume(GoogleServicesActivity.java:121)
       at com.heyya.installed.activity.abstracts.MainUIActivity.onResume(MainUIActivity.java:596)
       at com.heyya.installed.activity.MainActivity.onResume(MainActivity.java:143)
       at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1412)
       at android.app.Activity.performResume(Activity.java:7292)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3776)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3816)
       at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:193)
       at android.app.ActivityThread.main(ActivityThread.java:6669)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

It works ok on all other android versions

**Comment by justTalk** [comment]
Caused by java.lang.IllegalStateException Not allowed to start service Intent { cmp=com.x.x.x/.x.x.xService }: app is in background uid UidRecord
Caused by java.lang.IllegalStateException
Not allowed to start service Intent { cmp=com.x.x.x/.x.x.xService }: app is in background uid UidRecord{7d9f297 u0a184 SVC bg:+15m25s900ms idle change:uncached procs:3 seq(0,0,0)}
 
Android 8.0 的应用尝试在不允许其创建后台服务的情况下使用 startService() 函数，则该函数将引发一个 IllegalStateException。 新的 Context.startForegroundService() 函数将启动一个前台服务。现在，即使应用在后台运行， 系统也允许其调用 Context.startForegroundService()。不过，应用必须在创建服务后的五秒内调用该服务的 startForeground() 函数。
 
解决方法：
1. 修改启动方式

if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
    context.startForegroundService(intent);
} else {
    context.startService(intent);
}
2. 并且在service里再调用startForeground方法，不然就会出现ANR

context.startForeground(SERVICE_ID, builder.getNotification());

**Comment by jkasten2** [comment]
@bshka Thanks for the details. We removed `com.onesignal.OneSignal.startSyncService` in our 3.8.2 so it looks like your project pulled in an older version of the OneSignal SDK.

I recommend changing `implementation 'com.onesignal:OneSignal:[3.7.1, 3.99.99]'` to `implementation 'com.onesignal:OneSignal:[3.10.1, 3.99.99]'` and re-sync Gradle through Android Studio to get the latest version.

@justTalk Thanks for the suggestion but we have already switched this to a `JobService` in PR #470 for Android 5.0 and newer devices.
